### ISSUE TITLE
No reporting on synced AD users

### ISSUE THREAD

**Issue by WesSec** [jsa2/caOptics#10]
Hi there, first of all, thanks for your tool.

I am running the command as:

`node ./ca/main.js --mapping --allTerminations --clearPolicyCache --clearMappingCache` 

Unfortunately the tool only reports on accounts created in AAD. I use ADsync with synced hashes, so all regular users exist in AAD (the domain differs tho because of tiered the accounts, but they exist in the same tenant and the admin account I run caOptics with has the rights to read them).

Is there any way to report on the complete set of users?

**Comment by jsa2** [comment]
Hi and thx for posting an issue! - There should be no filtering on synced accounts, so this seems bit strange, but I will reply back once I have my assessment ready

**Comment by jsa2** [comment]
And just to clarify, the tool will only report on users, who are added via groups or explicit userId's - Any other gap is otherwise reported by the 'All' users condition (In case the gap is not covered by policy 'All') - This is very evident in environments, that have only include based policies in user conditions

**Comment by jsa2** [comment]
You can also DM me in Twitter @santasalojoosua, if you want to setup debugging session 1:1 - Always glad to help, and receive valuable feedback

**Comment by jsa2** [comment]
Hi, I used the same parameters, and excluded synced user from policy, to ensure that it would then show up in gaps. I attached example pics of the setup.

It could be some other bug, but I am not able reproduce it, so I will close this issue for now. As proposed, we can have 1:1 session to pinpoint / debug the issue.

![image](https://user-images.githubusercontent.com/58001986/204572426-57062349-5db1-46ad-b497-5f9454c06b06.png)

![image](https://user-images.githubusercontent.com/58001986/204572486-48e47c74-410b-4b9f-9d04-f22c3d04c81c.png)
